journalpump/journalpump.py

```diff
@@ -675,6 +690,9 @@ def process(self) -> SingleMessageReadResult:
         if not self.pump.check_match(new_entry):
             return SingleMessageReadResult(has_more=True, bytes_read=None)
```

Comment:
I guess we probably shouldn't remove this since it's existing behaviour, but I guess we should update the docs to indicate this is a top level config as well as a reader level one? And/or just log a deprecation message when top-level is configured. Can be a followup PR.